File: Problems/hacker_rank/Search/20150122_Circle_City/Solution.java
import java.util.Scanner;

public class Solution {
  public static void main(final String[] args)  {
    Scanner sc = new Scanner(System.in);
    final int t = Integer.parseInt(sc.nextLine());
    for ( int i = 0; i < t; ++i ) {
      String[] inps = sc.nextLine().split(" ");
      final int r = Integer.parseInt(inps[0]);
      final int k = Integer.parseInt(inps[1]);
      final double radius = Math.sqrt(r);
      int cnt = 0;
      for ( int x = 0; x < radius; ++x )  {
        double dy = Math.sqrt(r - x * x);
        int y = (int) dy;
        if ( y - dy == 0 )  {
          cnt += 4;
        }
      }
      System.out.println((cnt <= k ? "" : "im") + "possible");
    }
  }
}

File: Problems/hacker_rank/Search/20150122_Circle_City/Solution.scala
object Solution {
  def main(args: Array[String]) {
    val inps = io.Source.stdin.getLines.toArray
    val t = inps(0).toInt
    for ( i <- 1 to t ) {
      val ar = inps(i).split(" ").map((s: String) => s.toInt).toArray
      val r = ar(0)
      val k = ar(1)
      val radius = Math.sqrt(r)
      val end = if ( radius - radius.toInt == 0 ) radius.toInt else radius.toInt + 1
      val cnt = 4 * (0 until end).map((x: Int) => Math.sqrt(r - x * x)).filter((y: Double) => y - y.toInt == 0).size
      if ( cnt <= k )
        println("possible")
      else
        println("impossible")
    }
  }
}
